s5_base/run.sh

```diff
@@ -6,148 +6,284 @@
+    done
 fi
 
 if [ $stage -le 3 ]; then
-  echo "Prepare dict and lang directories"
-  local/prep_samromur_lang.sh data/train/text data
+    echo "Identify OOV words"
+    cut -d' ' -f2- data/train/text \
+    | tr ' ' '\n' | sort -u | grep -Ev '^$' \
+    > data/train/wordlist.txt || exit 1;
+    
+    comm -23 data/train/wordlist.txt \
+    <(cut -f1 $prondict_orig | sort -u) \
+    > data/train/oov_wordlist.txt || exit 1;
+    
+    echo "Use a grapheme-to-phoneme model to generate the pronunciation of OOV words in the training set"
+    g2p.py --apply data/train/oov_wordlist.txt --model $g2p_model \
+    --encoding="UTF-8" > data/train/oov_with_pron.txt || exit 1;
+    
+    echo "Add the OOV words to the prondict"
+    cat $prondict_orig data/train/oov_with_pron.txt | sort -k1,1 | uniq > "$prondict" || exit 1;
+    
+    echo "Calculate the OOV rate in the dev and test set after"
+    echo "incorporating the training text vocabulary into the lexicon"
+    for n in eval dev; do
+        nb_tokens=$(cut -d' ' -f2- data/$n/text | wc -w)
+        cut -d' ' -f2- data/$n/text \
+        | tr ' ' '\n' | sort |uniq -c \
+        > data/$n/words.cnt || exit 1;
+        
+        comm -23 <(awk '$2 ~ /[[:print:]]/ { print $2 }' data/$n/words.cnt | sort) \
+        <(cut -f1 $prondict | sort -u) > data/$n/vocab_text_only.tmp
+        nb_oov=$(join -1 1 -2 1 data/$n/vocab_text_only.tmp <(awk '$2 ~ /[[:print:]]/ { print $2" "$1 }' data/$n/words.cnt | sort -k1,1) \
+        | sort | awk '{total = total + $2}END{print total}')
+        oov=$(echo "scale=3;$nb_oov/$nb_tokens*100" | bc)
+        echo "The out of vocabulary rate for $n is: $oov" >> oov_rate || exit 1;
+    done
 fi
 
 if [ $stage -le 4 ]; then
-  echo "Create 3gram language model from kenlm"
-  cat data/train/text | cut -d' ' -f2- | sed -e \
- 's/[,.?!]//g' > data/text_lm_training.txt
-
-  mkdir -p data/lang_3gsmall
+    if [ ! -d data/lang ]; then
+        echo "Create the lexicon"
+        [ -d $localdict ] && rm -r $localdict
+        mkdir -p $localdict data/lang/log
+        $train_cmd --mem 4G data/lang/log/prep_lang.log \
+        local/prep_lang.sh \
+        $prondict        \
+        $localdict   \
+        data/lang
+    fi
+fi
 
-  # language  modeling files are typically in data/lang_Xgram
-  $big_memory_cmd logs/make_LM_3gsmall.log local/make_LM.sh --order 3 \
-  --small true --carpa false data/text_lm_training.txt data/lang/ \
-  data/local/dict/lexicon.txt data/lang_3gsmall
+if [ $stage -le 5 ]; then
+    echo "Preparing a pruned trigram language model"
+    mkdir -p data/log
+    $train_cmd --mem 24G data/log/make_LM_3g.log \
+    local/make_LM.sh \
+    --order 3 --carpa false \
+    --min1cnt 20 --min2cnt 10 --min3cnt 2 \
+    $lm_train data/lang \
+    $localdict/lexicon.txt data \
+    || error 1 "Failed creating a pruned trigram language model"
+    
+    echo "Preparing an unpruned 4g LM"
+    $train_cmd --mem 32G data/log/make_LM_4g.log \
```

Comment:
```suggestion
    $big_memory_cmd data/log/make_LM_4g.log \
```
Would this work with just 24G?